internal/pkg/pkg.go

```diff
@@ -77,23 +80,57 @@ func New(path string) (*Pkg, error) {
 
 // Kptfile returns the Kptfile meta resource by lazy loading it from the filesytem.
 // A nil value represents an implicit package.
-func (p *Pkg) Kptfile() *kptfilev1alpha2.KptFile {
-	if !p.kptfileLoaded {
-		// TODO
-		// p.kptfile = ...
-		p.kptfileLoaded = true
+func (p *Pkg) Kptfile() (*kptfilev1alpha2.KptFile, error) {
```

Comment:
Is this used in the hydration code?

Reply:
Actually wasn't being used earlier in hydration code, but added in this commit to ensure that `hydration` is invoked on valid kpt packages. Valid kpt package will have `Kptfile`. I am on the fence if I want to make this `invariant` part of `pkg.New()` constructor.

Reply:
Are you thinking about the commands such as `kpt pkg init` where the Kptfile does not exist?

Reply:
No, my comment was more about checking if `Kptfile` is missing in `pkg.New(...)` constructor.